src/DIRAC/ConfigurationSystem/Client/Utilities.py

```diff
@@ -578,3 +569,133 @@ def getDIRACGOCDictionary():
+    if not result['OK']:
+      return result
+    clients = result['Value']
+
+  for cliName, cliDict in clients.items():
+    cliDict['issuer'] = cliDict.get('issuer', getAuthAPI())
+    cliDict['authority'] = cliDict.get('authority', getAuthAPI())
+    if cliName == 'DIRACCLI':
+      if not cliDict.get('client_metadata'):
+        cliDict['client_metadata'] = {'response_types': ['device'],
+                                      'grant_types': ['urn:ietf:params:oauth:grant-type:device_code']}
+    elif cliName == 'WEBAPPDIRACCLI':
+      cliDict['token_endpoint_auth_method'] = cliDict.get('token_endpoint_auth_method', 'client_secret_basic')
+      if not cliDict.get('client_metadata'):
+        cliDict['client_metadata'] = {'response_types': ['code', 'id_token token', 'token'],
+                                      'redirect_uris': [cliDict['redirect_uri']],
+                                      'token_endpoint_auth_method': cliDict['token_endpoint_auth_method'],
+                                      'grant_types': ['device', 'authorization_code', 'refresh_token',
+                                                      'urn:ietf:params:oauth:grant-type:token-exchange']}
+    if clientName and clientName == cliName:
+      return S_OK(cliDict)
+
+    if clientID and clientID == cliDict['client_id']:
+      return S_OK(cliDict)
+  return S_OK({} if clientID else clients)
+
+
+def getAuthorisationServerMetadata():
+  """ Get authoraisation server metadata
+
+      :return: S_OK(dict)/S_ERROR()
+  """
+  path = '/Systems/Framework/%s/APIs/Auth' % getSystemInstance("Framework")
+  result = gConfig.getSections(path)
+  if not result['OK']:
+    return result
+
+  data = {}
+  if 'AuthorizationServer' in result['Value']:
+    result = gConfig.getOptionsDictRecursively('%s/AuthorizationServer' % path)
+    if not result['OK']:
+      return result
+    data = result['Value']
+
+  data['issuer'] = data.get('issuer', getAuthAPI())
+  if not data['issuer']:
+    return S_ERROR('Cannot found the Auth RESTful API base URL in the configuration.')
+  data['jwks_uri'] = data.get('jwks_uri', data['issuer'] + '/jwk')
+  data['token_endpoint'] = data.get('token_endpoint', data['issuer'] + '/token')
+  data['userinfo_endpoint'] = data.get('userinfo_endpoint', data['issuer'] + '/userinfo')
+  data['registration_endpoint'] = data.get('registration_endpoint', data['issuer'] + '/register')
+  data['authorization_endpoint'] = data.get('authorization_endpoint', data['issuer'] + '/authorization')
+  data['grant_types_supported'] = data.get('grant_types_supported', [
+      'code', 'authorization_code', 'urn:ietf:params:oauth:grant-type:device_code', 'refresh_token'
+  ])
+  data['response_types_supported'] = data.get('response_types_supported', [
+      'code', 'device', 'id_token token', 'id_token', 'token'
+  ])
+  data['code_challenge_methods_supported'] = data.get('code_challenge_methods_supported', ['S256'])
```

Comment:
Yes, its come from [Authorization Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414#section-2).
Endpoints are defined by `FrameworkSystem.API.AuthHandler` methods.
it will support two types of authorization, [DeviceFlow](https://datatracker.ietf.org/doc/html/rfc8628#section-1) and [AuthorizationCode](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1).